src/NumericalAlgorithms/Spectral/SwshTags.hpp

```diff
@@ -0,0 +1,204 @@
+/// in tags
+struct EthbarEthbar {};
+/// \brief Struct which acts as a placeholder for a spin-weighted derivative
+/// label in the spin-weighted derivative utilities, but represents a 'no-op':
+/// no derivative is taken.
+struct NoDerivative {};
+
+namespace detail {
+
+// utility function for determining the change of spin after a spin-weighted
+// derivative has been applied.
+template <typename DerivativeKind>
+constexpr int derivative_spin_weight_adjustment = 0;
+
+template <>
+constexpr int derivative_spin_weight_adjustment<Eth> = 1;
+template <>
+constexpr int derivative_spin_weight_adjustment<Ethbar> = -1;
+template <>
+constexpr int derivative_spin_weight_adjustment<EthEth> = 2;
+template <>
+constexpr int derivative_spin_weight_adjustment<EthbarEthbar> = -2;
+
+// The below tags are used to find the new type represented by the spin-weighted
+// derivative of a spin-weighted quantity. The derivatives alter the spin
+// weights, and so the utility `adjust_spin_weight_t<Tag, DerivativeKind>` is a
+// metafunction which determines the correct spin-weighted type for the
+// spin-weighted derivative `DerivativeKind` of `Tag`.
+//
+// if there is no `Tag::type::spin`, but the internal type is a compatible
+// complex vector type, the spin associated with `Tag` is assumed to be 0.
+template <typename DataType, typename DerivativeKind>
+struct adjust_spin_weight {
+  using type =
+      Scalar<SpinWeighted<DataType,
+                          derivative_spin_weight_adjustment<DerivativeKind>>>;
+};
+
+// case for if there is a `Tag::type::spin`
+template <typename DataType, int Spin, typename DerivativeKind>
+struct adjust_spin_weight<SpinWeighted<DataType, Spin>, DerivativeKind> {
+  using type = Scalar<SpinWeighted<
+      DataType, Spin + derivative_spin_weight_adjustment<DerivativeKind>>>;
+};
+
+template <typename Tag, typename DerivativeKind>
+using adjust_spin_weight_t =
+    typename adjust_spin_weight<typename Tag::type::type, DerivativeKind>::type;
+
+// Helper function for creating an appropriate prefix tag name from one of the
+// spin-weighted derivative types and an existing tag name
+template <typename DerivativeKind>
+std::string compose_spin_weighted_derivative_name(
+    const std::string& suffix) noexcept;
+
+}  // namespace detail
+
+/// \ingroup SwshGroup
+/// \brief Prefix tag representing the spin-weighted derivative of a
+/// spin-weighted scalar.
+///
+/// Template Parameters:
+/// - `Tag`: The tag to prefix
+/// - `DerivativeKind`: The type of spin-weighted derivative. This may be any of
+///   the labeling structs: `Eth`, `Ethbar`, `EthEth`, `EthbarEth`, `EthEthbar`,
+///   `EthbarEthbar`, or `NoDerivative`.
+///
+/// Type Aliases and static values:
+/// - `type`: Always a `SpinWeighted<Scalar<ComplexDataVector>, Spin>`, where
+///   `Spin` is the spin weight after the derivative `DerivativeKind` has been
+///   applied.
+/// - `tag`: An alias to the base tag `Tag`. Provided for applicability to
```

Comment:
please use `wrapped tag` instead of `base tag` since the terminology `base tag` is already in use for something different